controllers/machinedeletionremediation_controller_test.go

```diff
@@ -456,18 +457,115 @@ var _ = Describe("Machine Deletion Remediation CR", func() {
 				})
 			})
 		})
+
+		Context("Support to MHC created CR", func() {
+			When("Machine's node exists", func() {
+				BeforeEach(func() {
+					underTest = createRemediationOwnedByMHC(workerNode, workerNodeMachine)
+				})
+
+				It("MHC worker machine is deleted", func() {
```

Comment:
Nit: The two new unit tests have more assertions than before. Maybe use steps (using `by`) for easier understanding of test logic from code and mostly from reviewing it in a failure.

Reply:
Why do you think so? The test is exactly the same as "worker node remediation exists"

Reply:
Correct, and I have missed that they both share the same logic. I meant that these two new unit tests have more assertions than *most* of the unit tests.

Reply:
The additional logic is to mock the reprovision of the node, it is not related to the test, that's why I didn't think using `By` here is necessary.

However, you made me realize that the second test ("Machine's node does not exist" in MHC Context) shouldn't need to mock the reprovision of the node. I'll fix it